src/index.js

```diff
@@ -138,10 +138,15 @@ class Dropzone extends React.Component {
     this.isFileDialogActive = false;
 
     fileList.forEach((file) => {
-      if (this.fileAccepted(file) && this.fileMatchSize(file)) {
-        if (!disablePreview) {
+      if (!disablePreview) {
```

Comment:
A cool idea. 

For my own selfish reasons–we have a feature release Friday that's waiting on this bug fix–I'm not super eager to take on this behaviour at the moment.

It could be part of a follow up PR though!